custommacros.sty

```diff
@@ -11,6 +11,7 @@
 \makeatletter
 \def\app@exe{\immediate\write18}
 \def\inputAllFiles#1{%
+  \app@exe{rm -f #1/\jobname.tmp}
   \app@exe{ls #1/*.tex | xargs cat >> #1/\jobname.tmp}%
```

Comment:
How about making it simpler which does not rely on `xargs` and append mode, but concatenate everything together and overwrite the temporary file each time?

```suggestion
  \app@exe{cat #1/*.tex > #1/\jobname.tmp}%
```